go/src/vendors/sonic/client/dbclient_test/counter_repository_test.go

```diff
@@ -0,0 +1,215 @@
+	ethernet3_redis_key             = "COUNTERS:oid-3"
+	HGetAllMethod                   = "HGetAll"
+	HmGetMethod                     = "HmGet"
+	SAI_PORT_STAT_IF_IN_ERRORS      = "SAI_PORT_STAT_IF_IN_ERRORS"
+	SAI_PORT_STAT_IF_IN_UCAST_PKTS  = "SAI_PORT_STAT_IF_IN_UCAST_PKTS"
+	SAI_PORT_STAT_IF_OUT_UCAST_PKTS = "SAI_PORT_STAT_IF_OUT_UCAST_PKTS"
+	COUNTERS_PORT_NAME_MAP          = "COUNTERS_PORT_NAME_MAP"
+)
+
+func getInterfaceToODIMapping() map[string]string {
+	newMap := map[string]string{
+		ethernet1: oid1,
+		ethernet2: oid2,
+		ethernet3: oid3,
+	}
+	return newMap
+}
+
+func getCountersForInterfaces(strs []string) []interface{} {
+	counters := make([]interface{}, len(strs))
+	for i, s := range strs {
+		counters[i] = s
+	}
+	return counters
+}
+
+func validateOrderOfFields(fields []string) bool {
+	return fields[0] == SAI_PORT_STAT_IF_IN_ERRORS && fields[1] == SAI_PORT_STAT_IF_IN_UCAST_PKTS && fields[2] == SAI_PORT_STAT_IF_OUT_UCAST_PKTS
+}
+
+type MockRedisProvider struct {
+	mock.Mock
+}
+
+func (mockRedisProvider *MockRedisProvider) HmGet(database int, key string, fields []string) ([]interface{}, error) {
+	args := mockRedisProvider.Called(database, key, fields)
+	return args.Get(0).([]interface{}), args.Error(1)
+}
+
+func (mockRedisProvider *MockRedisProvider) HGetAll(database int, key string) (map[string]string, error) {
+	args := mockRedisProvider.Called(database, key)
+	return args.Get(0).(map[string]string), args.Error(1)
+}
+
+func Test_GetInterfaceCounters_ReturnsAllCountersSuccessfuly(t *testing.T) {
+	mockRedisProvider := new(MockRedisProvider)
+	newMap := getInterfaceToODIMapping()
+	(mockRedisProvider).On(HGetAllMethod, 2, COUNTERS_PORT_NAME_MAP).Return(newMap, nil)
+
+	strs := []string{"110", "111", "112"}
+	counters := getCountersForInterfaces(strs)
+
+	mockRedisProvider.On(HmGetMethod, 2, ethernet1_redis_key, mock.MatchedBy(validateOrderOfFields)).Return(counters, nil)
+
+	strs1 := []string{"210", "211", "212"}
+	counters1 := getCountersForInterfaces(strs1)
+	mockRedisProvider.On(HmGetMethod, 2, ethernet2_redis_key, mock.MatchedBy(validateOrderOfFields)).Return(counters1, nil)
+
+	strs2 := []string{"310", "311", "312"}
+	counters2 := getCountersForInterfaces(strs2)
+	mockRedisProvider.On(HmGetMethod, 2, ethernet3_redis_key, mock.MatchedBy(validateOrderOfFields)).Return(counters2, nil)
+	counterDBClient := CounterRepository{RedisProvider: mockRedisProvider}
+	result, err := counterDBClient.GetInterfaceCounters()
+	mockRedisProvider.AssertNumberOfCalls(t, HmGetMethod, 3)
+	mockRedisProvider.AssertExpectations(t)
+	assert := assert.New(t)
+	assert.NotEqual(nil, result, "Result is expected to be non nil")
+	assert.Equal(nil, err, "GetInterfaceCounters: Error is expected to be nil")
+	assert.Equal(3, len(result), "GetInterfaceCounters: length of resulting map is expected to be 3")
+	assert.Equal(uint64(110), result[ethernet1]["IfInErrors"], "Ehternet1's IfInErrors counter is expected to be 110")
```

Comment:
This is good.

Next time, you may think of going data driven.

e.g.
create the i/f counters return value as i/p to mock and just one line comparison of o/p. Something in this line